samples/hal/lapack_hal/impl.cpp

```diff
@@ -0,0 +1,454 @@
+#include "impl.hpp"
```

Comment:
@sovrasov, @mshabunin, I think, this part should be somehow embedded into core, i.e. if user specifies some WITH_LAPACK flag (or rather does not specify WITH_LAPACK=OFF), it should try to detect BLAS/LAPACK and in the case of success make use of it; of course, it should be done via overriding the same HAL calls, i.e. hal::SVD32f body should call lapack_svd when LAPACK is found